.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Main workflow 
+
+on: [push, pull_request]
```

Comment:
I think we want something like this so we don't build redundantly only non-master pushed branches (but we still build on PRs):

```yaml
on:
  - push:
      branches:
        - master
  - pull_request
```

Reply:
This didn't work so I ended up duplicating the workflow to have a separate one for master on push for now

Reply:
This should work:

```yaml
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```